src/tokens/saga.ts

```diff
@@ -309,6 +325,49 @@ export function* watchAccountFundedOrLiquidated() {
   }
 }
 
+export async function fetchImportedTokenBalances(
+  address: Address,
+  importedTokens: TokenBalance[],
+  knownTokenBalances: Record<string, FetchedTokenBalance>
+) {
+  const importedTokensWithBalance: StoredTokenBalances = {}
+
+  const balanceRequests = importedTokens.map(async (importedToken) => {
+    try {
+      if (!importedToken) {
+        return
+      }
+
+      let fetchedBalance
+      if (knownTokenBalances[importedToken.tokenId]) {
+        fetchedBalance = knownTokenBalances[importedToken.tokenId].balance
+      } else {
+        const contract = getContract({
+          abi: erc20.abi,
+          address: importedToken!.address as Address,
+          client: {
+            public: publicClient[networkIdToNetwork[importedToken.networkId]],
+          },
+        })
+        fetchedBalance = (await contract.read.balanceOf([address])).toString()
+      }
+
+      const balance = new BigNumber(fetchedBalance).shiftedBy(-importedToken.decimals).toFixed()
+
+      importedTokensWithBalance[importedToken.tokenId] = {
+        ...importedToken,
+        balance,
+        priceUsd: undefined,
+      }
+    } catch (error) {
+      Logger.error(TAG, 'Error fetching imported token balance', error)
```

Comment:
```suggestion
      Logger.error(TAG, `Error fetching imported token balance with address ${importedToken?.address}` error)
```
(in case the error does not mention this?)